app/web/features/auth/password/CompleteResetPassword.tsx

```diff
@@ -1,58 +1,151 @@
-import { useEffect } from "react";
+import stringOrFirstString from "utils/stringOrFirstString";
+import { useAuthContext } from "features/auth/AuthProvider";
+import { Typography, Container } from "@material-ui/core";
+import StyledLink from "components/StyledLink";
+import TextField from "components/TextField";
+import HtmlMeta from "components/HtmlMeta";
 import { useMutation } from "react-query";
+import { useForm } from "react-hook-form";
+import makeStyles from "utils/makeStyles";
+import { useRouter } from "next/router";
+import { AUTH } from "i18n/namespaces";
+import Button from "components/Button";
+import { useTranslation } from "i18n";
+import Alert from "components/Alert";
 import { loginRoute } from "routes";
+import { RpcError } from "grpc-web";
 import { service } from "service";
-import stringOrFirstString from "utils/stringOrFirstString";
+
+const useStyles = makeStyles((theme) => ({
+  form: {
+    "& > * + *": {
+      marginBlockStart: theme.spacing(1),
+    },
+  },
+  standardContainer: {
+    marginTop: theme.spacing(2),
+    paddingLeft: theme.spacing(2),
+    paddingRight: theme.spacing(2),
+    paddingBottom: theme.spacing(2),
+    flex: 1,
+  },
+  main: {
+    padding: theme.spacing(0, 3),
+  },
+  textField: {
+    "& > div": {
+      width: "100%",
+      marginBottom: theme.spacing(2),
+      [theme.breakpoints.up("md")]: {
+        width: theme.typography.pxToRem(400),
+      },
+    },
+  },
+}));
 
 export default function CompleteResetPassword() {
+  const { authState } = useAuthContext();
   const { t } = useTranslation(AUTH);
-  const classes = useAppRouteStyles();
+  const formClass = useStyles();
+  const { handleSubmit, register } =
+    useForm<{ newPassword: string; newPasswordCheck: string }>();
 
   const router = useRouter();
   const resetToken = stringOrFirstString(router.query.token);
+  const isResetTokenOk =
+    !!resetToken && typeof resetToken === "string" && resetToken !== ""; // TODO: needed??
 
-  const {
-    error,
-    isLoading,
-    isSuccess,
-    mutate: completePasswordReset,
-  } = useMutation<Empty, RpcError, string>((resetToken) =>
-    service.account.completePasswordReset(resetToken)
+  const { error, isLoading, isSuccess, mutate } = useMutation<
+    Empty,
+    RpcError,
+    string
+  >((newPassword) =>
+    service.account.CompletePasswordResetV2(resetToken as string, newPassword)
   );
 
-  useEffect(() => {
-    if (resetToken) {
-      completePasswordReset(resetToken);
+  const onSubmit = handleSubmit(({ newPassword, newPasswordCheck }) => {
+    if (newPassword !== newPasswordCheck) {
+      alert(t("change_password_form.password_mismatch_error"));
+      return;
     }
-  }, [completePasswordReset, resetToken]);
+
+    mutate(newPassword);
+  });
+
+  // TODO: needed?
+  if (authState.authenticated) {
+    return (
+      <Container className={formClass.standardContainer}>
+        <Alert severity="error">
+          Can't changed the password if you are logged in
```

Comment:
This should be a translation string, right? Also add "You can change your password in account settings, or log out to reset your password." or similar

Reply:
true!, I forgot to change it....

Reply:
ah I remember I put a comment because I was unsure if I should keep this code snippet or not, I think github deleted the comment when I updated the file

Reply:
![image](https://github.com/Couchers-org/couchers/assets/16777161/cd89e145-9751-433b-8b1c-ad6bb00ea26a)